pkg/asset/installconfig/azure/validation.go

```diff
@@ -75,7 +75,16 @@ func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceTyp
 			}
 			if cpus < float64(req.minimumVCpus) {
 				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUs", req.minimumVCpus)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type", "vCPUs"), instanceType, errMsg))
+			}
+		} else if strings.EqualFold(*capability.Name, "vCPUsAvailable") {
```

Comment:
We should be checking `vCPUsAvailable` **instead** of `vCPUs` rather than in addition to.

Reply:
See https://docs.microsoft.com/en-us/azure/virtual-machines/constrained-vcpu.
Essentially, the number of available vCPUs will always be less than or equal to the number of vCPUs for an instance type.